### PR TITLE
keep the index stable on equal timestamps

### PR DESCRIPTION
**Pull request by defiantnerd**
when sorting the events by timestamp, an item must be sorted to front 
if the timestamp if event[a] is earlier than the timestamp of event[b].

if they have the same timestamp, the index must be preserved, so notes (with a note-id) are before note expressions etc.

### CODE REVIEW

File: src/detail/vst3/process.cpp [hunk 1]
```diff
@@ -460,9 +460,20 @@ bool ProcessAdapter::output_events_try_push(const struct clap_output_events* lis
...
+  // if they have the same timestamp, the index must be preserved
+
   std::sort(_eventindices.begin(), _eventindices.end(),
             [&](size_t const& a, size_t const& b)
-            { return _events[a].header.time < _events[b].header.time; });
+            {
+              auto t1 = _events[a].header.time;
+              auto t2 = _events[b].header.time;
+              if (t1 < t2) return true;
+              if (t1 == t2) return (a < b);
```
[the remark this answers is not in the document]
Comment (defiantnerd) [reply src/detail/vst3/process.cpp:474]: Yes, I thought I should put the „more likely“ code in front, but actually you‘re right. This is micro optimization and I don‘t know what the more likely case is. I‘ll take your variant until we can use [[likely]].

[the remark this answers is not in the document]
Comment (baconpaul) [reply src/detail/vst3/process.cpp:474]: readability with the more likely in front is higher i agree but fine either way.

**Review by baconpaul — APPROVED** [review]

File: src/detail/vst3/process.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (0ax1) [reply src/detail/vst3/process.cpp:474]: Either is fine by me.

**Review by 0ax1 — APPROVED** [review]

File: src/detail/vst3/process.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (0ax1) [reply src/detail/vst3/process.cpp:474]: To be clear, the branch predictor doesn't care about the order here.

[the remark this answers is not in the document]
Comment (defiantnerd) [reply src/detail/vst3/process.cpp]: tested best version with godbolt compiler explorer. :)

File: src/detail/vst3/process.cpp [hunk 2]
```diff
@@ -460,9 +460,18 @@ bool ProcessAdapter::output_events_try_push(const struct clap_output_events* lis
...
+  // the timestamp of event[b].
+  // if they have the same timestamp, the index must be preserved
+
   std::sort(_eventindices.begin(), _eventindices.end(),
             [&](size_t const& a, size_t const& b)
-            { return _events[a].header.time < _events[b].header.time; });
+            {
+              auto t1 = _events[a].header.time;
+              auto t2 = _events[b].header.time;
+              return (t1 == t2) ? (a < b) : (t1 < t2);
```
Comment (0ax1) [inline src/detail/vst3/process.cpp:473]: Great attention to detail here checking back the assembly!

**Pull request merged**